manifests/kustomize/base/pipeline/ml-pipeline-semaphore-configmap.yaml

```diff
@@ -0,0 +1,5 @@
+kind: ConfigMap
```

Comment:
I think this lgtm, but I just had a thought. What happens if:
1. I install kfp and I get this configmap created
2. I customize it by adding keys to it
3. I upgrade to the next version of kfp

Is that down the line upgrade going to overwrite my customized configmap with this blank one? It'd be cool if we could test that somehow -- perhaps by manually creating the configmap on a cluster, putting data in it, and then installing kfp.

Reply:
Makes sense. 
To resolve this, I've added a job to the configmap, `semaphore-configmap-init` that runs during the Kustomize deployment process.
It checks for the existence of the semaphore-config ConfigMap using kubectl get.
If the ConfigMap does not exist, it creates one with an empty init key.
If the ConfigMap already exists, the Job skips creation and exits successfully.